lib/util/git.js

```diff
@@ -25,6 +26,23 @@ const GOOD_ENV_VARS = new Set([
   'GIT_SSL_NO_VERIFY'
 ])
 
+const GIT_TRANSIENT_ERRORS = [
+  'remote error: Internal Server Error',
+  'fatal: Couldn\'t find remote ref ',
```

Comment:
I think a failure to find a remote ref is very unlikely to be fixed by an immediate retry -- even if there's a timing issue, it's unlikely to be fixed in such a small window, and this particular failure, I would expect, is almost always an error the user themself has to fix.

Reply:
agree